File: src/TlkStringRef.java
import java.io.FileInputStream;
import java.io.IOException;

public class TlkStringRef {
    public int stringId;
    public int BitOffset;

    public String Data;
    public int position;

    public TlkStringRef(FileInputStream r) throws IOException {
        this.stringId = r.read();
        this.BitOffset = r.read();
    }
}

File: python/tlk_string_ref.py
import input_stream


class TlkStringRef:
    def __init__(self, in_stream):
        self.string_id = input_stream.read_int_32(in_stream)
        self.bit_offset = input_stream.read_int_32(in_stream)
        self.data = None
        self.position = None
